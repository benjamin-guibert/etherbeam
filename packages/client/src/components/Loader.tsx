import React, { ReactElement } from 'react'
import './Loader.scss'

type LoaderTypeProp = 'spinner' | 'dots'

interface LoaderProps {
  type: LoaderTypeProp
}

const Loader = ({ type }: LoaderProps): ReactElement => {
  return (
    <div className="my-loader">
      <div className={`my-loader-${type}`}></div>
    </div>
  )
}

export default Loader
