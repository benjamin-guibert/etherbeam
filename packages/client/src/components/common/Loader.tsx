import React, { ReactElement } from 'react'
import Spinner from 'react-bootstrap/Spinner'

interface LoaderProps {
  size?: 'sm'
  className?: string
}

const Loader = ({ size, className }: LoaderProps): ReactElement => {
  return (
    <div className={['text-center', className].join(' ')}>
      <Spinner className="text-center m-1" variant="secondary" animation="border" role="status" size={size}>
        <span className="sr-only">Loading...</span>
      </Spinner>
    </div>
  )
}

export default Loader
