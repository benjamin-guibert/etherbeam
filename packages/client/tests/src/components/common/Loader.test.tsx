import * as React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import Loader from 'components/common/Loader'

describe('<Loader />', () => {
  let component: ReactWrapper

  beforeAll(() => {
    component = mount(<Loader size="sm" className="ae-class" />)
  })

  it('should render', () => {
    expect(component.find('div').first().hasClass('ae-class')).toBeTruthy()
    expect(component.find('Spinner').prop('size')).toBe('sm')
  })

  it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
})
