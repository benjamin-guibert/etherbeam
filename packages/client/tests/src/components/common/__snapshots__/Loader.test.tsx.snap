// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should match snapshot 1`] = `
<div
  class="text-center ae-class"
>
  <div
    class="text-center m-1 spinner-border spinner-border-sm text-secondary"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
